### PR TITLE
node: Fix panic when node store is not initialized yet

### PR DESCRIPTION
**Pull request by tgraf**
Because node store registration happens in the background, the nodeStore
pointer can still be nil at the time the local node is updated first.

Fixes: 4a7d484af44b ("Daemon: Update Health endpoints IP on localNode.")
Fixes: #5137

Signed-off-by: Thomas Graf <thomas@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5139)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: pkg/node/local_node.go [hunk 1]
```diff
@@ -73,5 +75,12 @@ func ConfigureLocalNode() error {
 // NotifyLocalNodeUpdated Update local node information in the key-value
 // storage
 func NotifyLocalNodeUpdated() error {
```
Comment (ianvernon) [inline pkg/node/local_node.go]: Not blocking, but you can remove the returned `error` because it is always `nil` with your change.

**Review by ianvernon — APPROVED** [review]

File: pkg/node/local_node.go [hunk 2]
```diff
@@ -15,14 +15,17 @@
...
 
+	"github.com/cilium/cilium/pkg/option"
+
 	"k8s.io/api/core/v1"
 )
 
-var localNode Node
+var (
+	localNode      Node
+	nodeRegistered = make(chan struct{})
```
Comment (rlenglet) [inline pkg/node/local_node.go:28]: This is now a global variable, so if `ConfigureLocalNode()` is called more than once, e.g. in unit tests, `close(nodeRegistered)` panics.

**Comment by tgraf** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Review by rlenglet — APPROVED** [review]

**Review by eloycoto — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Comment by joestringer** [comment]
Marked as needs-backport/1.2 as #5117 (which this fixes) is also marked that way.

**Pull request merged**